product_cost_price_avco_sync/models/stock_move.py

```diff
@@ -0,0 +1,101 @@
+            ]).unlink()
+
+    def get_average_price(self, previous_qty, previous_price):
+        new_std_price = ((previous_qty * previous_price) + (
+            self.price_unit * self.quantity_done)) / (previous_qty +
+                                                      self.quantity_done)
+        return new_std_price
+
+    def cost_price_avco_sync(self):
+        procesed_moves = set()
+        for move in self.sorted('date'):
+            if (move.id in procesed_moves or move.state != 'done' or
+                    not move._is_in() or
+                    move.product_id.cost_method != 'average'):
+                continue
+            previous_instant = move._previous_instant_date()
+            previous_qty = move.product_id.with_context(
+                to_date=previous_instant
+            ).qty_available
+            previous_price = move.product_id.get_history_price(
+                move.company_id.id, previous_instant) if previous_qty else 0.0
+            price = move.get_average_price(previous_qty, previous_price)
+            move._remove_after_history_price()
+            move._create_price_history(price)
+            previous_qty += move.quantity_done
+            affected_moves = self.with_context(skip_avco_sync=True).search([
+                ('product_id', '=', move.product_id.id),
+                ('date', '>', move.date),
+            ], order='date')
+            for af_move in affected_moves:
+                if af_move._is_in():
+                    if af_move.inventory_id or af_move.move_orig_ids:
+                        af_move.write({
+                            'price_unit': price,
+                            'value': price * af_move.quantity_done,
+                        })
+                    # Avoid reprocess move if included in self.write
+                    procesed_moves.add(af_move.id)
+                    price = af_move.get_average_price(previous_qty, price)
+                    af_move._create_price_history(price)
+                    previous_qty += af_move.quantity_done
+                elif af_move._is_out():
+                    af_move.write({
+                        'price_unit': -price,
+                        'value': -price * af_move.quantity_done,
+                    })
+                    previous_qty -= af_move.quantity_done
+                    # Avoid worst price because core don't works negatives fine
+                    if previous_qty < 0.0:
+                        previous_qty = 0.0
+            if float_compare(move.product_id.standard_price, price,
+                             precision_rounding=move.product_uom.rounding):
+                # Write the standard price, as SUPERUSER_ID because a warehouse
+                # manager may not have the right to write on products
+                move.product_id.with_context(
+                    force_company=move.company_id.id,
+                ).sudo().write({'standard_price': price})
```

Comment:
Is this duplicating latest cost price in history?

Reply:
It is possible, but it is something that usually occurs in odoo and that in v13 has changed